docs/experimental/explanation/metrics.md

```diff
@@ -0,0 +1,135 @@
+    
+```python
+from ragas_experimental.metric import numeric_metric
+
+@numeric_metric
+def my_metric(predicted: float, expected: float) -> float:
+    return abs(predicted - expected)
+```
+
+3. **Ranked Metrics**
+
+These evaluate multiple outputs at once and return a ranked list based on a defined criterion. They are useful when the goal is to compare outputs relative to one another.
+
+```python
+from ragas_experimental.metric import ranked_metric
+@ranked_metric
+def my_metric(predicted: list, expected: list) -> list:
+    return sorted(predicted, key=lambda x: x['score'], reverse=True)
+```
+
+
+## LLM-based vs. Non-LLM-based Metrics
+
+### Non-LLM-based Metrics
+
+These metrics are deterministic functions evaluating predefined inputs against clear, finite criteria.
+
+Example:
+
+```python
+def my_metric(predicted: str, expected: str) -> str:
+    return "pass" if predicted.lower() == expected.lower() else "fail"
+```
+
+When to use:
+
+- Tasks with strictly defined correct outcomes (e.g., mathematical solutions, deterministic tasks like booking agents updating databases).
+
+### LLM-based Metrics
+
+These leverage LLMs (Large Language Models) to evaluate outcomes, typically useful where correctness is nuanced or highly variable.
+
+Example:
+```python
+def my_metric(predicted: str, expected: str) -> str:
+    response = llm.generate(f"Evaluate semantic similarity between '{predicted}' and '{expected}'")
+    return "pass" if response > 5 else "fail"
+```
+
+When to use:
+
+- Tasks with numerous valid outcomes (e.g., paraphrased correct answers).
+- Complex evaluation criteria aligned with human or expert preferences (e.g., distinguishing “deep” vs. “shallow” insights in research reports). Although simpler metrics (length or keyword count) are possible, LLM-based metrics capture nuanced human judgment more effectively.
+
+## Choosing the Right Metrics for Your Application
+
+1. **Prioritize End-to-End Metrics**
+
+Focus first on metrics reflecting overall user satisfaction. While many aspects influence user satisfaction—such as factual correctness, response tone, and explanation depth—concentrate initially on the few dimensions delivering maximum user value (e.g., answer and citation accuracy in a RAG-based assistant).
+
+2. **Ensure Interpretability**
+
+Design metrics clear enough for the entire team to interpret and reason about. For example:
+
+- Execution accuracy in a text-to-SQL system: Does the SQL query generated return precisely the same dataset as the ground truth query crafted by domain experts?
+
+3. **Emphasize Objective Over Subjective Metrics**
+
+Prioritize metrics with objective criteria, minimizing subjective judgment. Assess objectivity by independently labeling samples across team members and measuring agreement levels. A high inter-rater agreement (≥80%) indicates greater objectivity.
+
+4. **Few Strong Signals over Many Weak Signals**
```

Comment:
these are subheadings - use h3 for those

this is in other areas as well - this defeats the purpose of markdown syntax